spec/features/class_member/creating_a_class_member_spec.rb

```diff
@@ -23,55 +18,76 @@
     }
   end
 
-  it 'responds 201 Created' do
-    post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
-    expect(response).to have_http_status(:created)
+  before do
+    authenticated_in_hydra_as(owner)
+    stub_user_info_api_for(student)
   end
 
-  it 'responds 201 Created when the user is a school-teacher' do
-    authenticated_in_hydra_as(teacher)
+  context 'with valid params' do
+    let(:student_attributes) { [{ id: student.id, name: student.name, username: student.username }] }
 
-    post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
-    expect(response).to have_http_status(:created)
-  end
+    before do
+      stub_profile_api_list_school_students(school:, student_attributes:)
+    end
 
-  it 'responds with the class member JSON' do
-    post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
-    data = JSON.parse(response.body, symbolize_names: true)
+    it 'responds 201 Created' do
+      post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
+      expect(response).to have_http_status(:created)
+    end
 
-    expect(data[:student_id]).to eq(student.id)
-  end
+    it 'responds 201 Created when the user is a school-teacher' do
+      authenticated_in_hydra_as(teacher)
 
-  it 'responds with the student JSON' do
-    post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
-    data = JSON.parse(response.body, symbolize_names: true)
+      post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
+      expect(response).to have_http_status(:created)
+    end
 
-    expect(data[:student_name]).to eq('School Student')
-  end
+    it 'responds with the class member JSON' do
+      post("/api/schools/#{school.id}/classes/#{school_class.id}/members", headers:, params:)
+      data = JSON.parse(response.body, symbolize_names: true)
 
-  it "responds with nil attributes for the student if their user profile doesn't exist" do
-    student_id = SecureRandom.uuid
-    new_params = { class_member: params[:class_member].merge(student_id:) }
+      student_ids = data.pluck(:student_id)
+      expect(student_ids).to eq([student.id])
+    end
```

Comment:
Wait, so posting to the single endpoint is still going to return a list (of one)? If so, this isn't what I expected

Reply:
and done 👍 had fun with jbuilder partials for this one!